source/diff/diff.cpp

```diff
@@ -292,6 +292,19 @@ class Differ {
       opt::IteratorRange<opt::Module::const_inst_iterator> src_insts,
       opt::IteratorRange<opt::Module::const_inst_iterator> dst_insts);
 
+  // Helper functions to group ids for matching.
+  std::string GroupIdsHelperGetName(const IdInstructions& id_to, uint32_t id);
+  uint32_t GroupIdsHelperGetTypeId(const IdInstructions& id_to, uint32_t id);
+  template <typename T>
+  void GroupIds(const IdGroup& ids, bool is_src, std::map<T, IdGroup>* groups,
+                std::function<T(const IdInstructions&, uint32_t)> get_group);
+  template <typename T>
+  void GroupIdsAndMatch(
```

Comment:
This function could also use some comment to describe what it does.  In particular, is should lay out what the functions `get_group` and `match_group` should do.

Reply:
Done.